### PR TITLE
[SYCLomatic] Update CodePin doc

### PR DESCRIPTION
**Pull request by AndyCHHuang**

### CODE REVIEW

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 1]
```diff
@@ -24,10 +24,8 @@ the CUDA and SYCL programs to help identify the source of divergent runtime beha
 Enable CodePin
 --------------
 
-Enable CodePin with the ``–enable-codepin`` option. If ``–out-root`` is specified,
-the instrumented CUDA program will be put into a folder with a ``_debug`` postfix
-beside the out-root folder. Otherwise, the instrumented CUDA program will be put
-in the default folder ``dpct_output_debug``.
+Enable CodePin with the ``–enable-codepin`` option. The instrumented CUDA program will be put
+in the folder ``dpct_output_codepin_cuda``.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]: what about instrumented SYCL code?

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 2]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
-                 },
-                 "d_a[3]": {
-                    "m_Data": "00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00"
-                 },
-                 "d_result[0]": {
-                    "m_Data": "00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00"
-                 },
+            [
+                {
+                    "ID": "/home/yyergg/workspace/simple_test/test.cu:84:3:prolog",
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
                    "ID": "test.cu:84:3:prolog",
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 3]
```diff
@@ -241,7 +239,8 @@ After migration, there will be two files: ``dpct_output/example.dp.cpp`` and ``d
     Result[3]: (2, 3, 4)
     */
 
-After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_debug/example.cu``, the following reports will be generated. Line number 13 shows the point of divergence.
+After building and executing ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu``,
+the following reports will be generated.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
the following execution log files will be generated.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 4]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
             ...
 
 The report helps identify where the runtime behavior of the CUDA and the SYCL
-programs start to diverge from one another.
\ No newline at end of file
+programs start to diverge from one another.
+
+Analyze the Data Checkpoints
+-------
+codepin-report.py is a tool consumes the data point files from both CUDA and SYCL and performs auto analysis of the data checkpoints.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
codepin-report.py is a tool consumes the execution log file generated from both CUDA and SYCL code and performs auto analysis.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 5]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+codepin-report.py can identify the in consistent data value and report the stats data of the data checkpoints.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
codepin-report.py can identify the in consistent data value and report the stats data of the execution log including data checkpoints.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 6]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+
+codepin-report.py consumes the data point files from both CUDA and SYCL with the following commandline.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
codepin-report.py consumes the execution log file generated from both CUDA and SYCL code with the following command line.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 7]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+``codepin-report.py [-h] --instrumented-cuda-log <file path> --instrumented-sycl-log <file path>``
+
+Following is an example of the analysis report.
+
+.. code-block::
+
+    CodePin Summary
+    Totally APIs count, 2
+    Consistently APIs count, 2
+    Most Time-consuming Kernel(CUDA), /home/yyergg/workspace/codepin_demo/example.cu:26:3:epilog, time:8.2316
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
    Most Time-consuming Kernel(CUDA), example.cu:26:3:epilog, time:8.2316
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 8]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+    Most Time-consuming Kernel(SYCL), /home/yyergg/workspace/codepin_demo/example.cu:26:3:epilog, time:10.2575
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
    Most Time-consuming Kernel(SYCL), example.cu:26:3:epilog, time:10.2575
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 9]
```diff
@@ -241,7 +239,8 @@ After migration, there will be two files: ``dpct_output/example.dp.cpp`` and ``d
     Result[3]: (2, 3, 4)
     */
 
-After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_debug/example.cu``, the following reports will be generated. Line number 13 shows the point of divergence.
+After building and executing ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu``,
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
After building ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu`` and executing the binaries built out
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 10]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
             ...
 
 The report helps identify where the runtime behavior of the CUDA and the SYCL
-programs start to diverge from one another.
\ No newline at end of file
+programs start to diverge from one another.
+
+Analyze the Data Checkpoints
+-------
+codepin-report.py is a tool consumes the execution log files from both CUDA and SYCL code and performs auto analysis.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
codepin-report.py (also can be triggered by dpct/c2s --codepin-report) is a functionality of the compatibility tool consumes the execution log files from both CUDA and SYCL code and performs auto analysis.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 11]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+codepin-report.py can identify the in consistent data value and report the stats data of the execution log including data checkpoints.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
codepin-report.py can identify the inconsistent data value and report the stats data of the execution.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 12]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+
+codepin-report.py consumes the execution log files from both CUDA and SYCL code with the following command line.
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]:
```suggestion
codepin-report.py consumes the execution log files generated from both CUDA and SYCL code with the following command line.
```

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 13]
```diff
@@ -252,46 +251,92 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+``codepin-report.py [-h] --instrumented-cuda-log <file path> --instrumented-sycl-log <file path>``
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]: --floating-point-comparison-epsilon

File: docs/dev_guide/migration/debug-with-codepin.rst (hunk 10 above)
Comment (lapel22) [reply docs/dev_guide/migration/debug-with-codepin.rst]: Missing a word: suggest adding the word 'that' to Zhiming's edit above: codepin-report.py (can also be triggered by dpct/c2s --codepin-report) is a functionality of the compatibility tool that consumes the execution log files from both CUDA and SYCL code and performs auto analysis.

**Review by lapel22 — CHANGES_REQUESTED** [review]
Reviewed and left a small suggestion. Other than that, this looks good to me

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 14]
```diff
@@ -241,7 +239,8 @@ After migration, there will be two files: ``dpct_output/example.dp.cpp`` and ``d
     Result[3]: (2, 3, 4)
     */
 
-After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_debug/example.cu``, the following reports will be generated. Line number 13 shows the point of divergence.
+After building ``dpct_output_codepin_sycl/example.dp.cpp`` and ``dpct_output_debug/example.cu`` and executing the binaries built out
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]: should be dpct_output_codepin_cuda/example.cu ?

File: docs/dev_guide/migration/debug-with-codepin.rst [hunk 15]
```diff
@@ -252,46 +251,109 @@ After building and executing ``dpct_output/example.dp.cpp`` and ``dpct_output_de
...
+                                        }
+                                    },
             ...
 
 The report helps identify where the runtime behavior of the CUDA and the SYCL
-programs start to diverge from one another.
\ No newline at end of file
+programs start to diverge from one another.
+
+Analyze the Data Checkpoints
```
Comment (zhimingwang36) [inline docs/dev_guide/migration/debug-with-codepin.rst]: L 322 should be updated as "Most Time-consuming Kernel" is not info of  Data Checkpoints . 
how about Analysis the  CodePin result?

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
Have addressed all comment.

**Pull request merged**